### DISCUSSION TITLE
Double and misplaced business

### DISCUSSION THREAD

**Discussion by SteaceP** [osmlab/name-suggestion-index#4887] [Q&A]
Hi,

I sometimes use osm and decided to contribute.
I started small to accommodate myself with the project. I tried to remove the warning at build time, I have noticed double/triple entry. Like the fast-food restaurants "Yoshinoya" has 3 entries with those ids:
yoshinoya-ffe475
yoshinoya-20e6f1
527fda-658eea
Some other businesses are not fast food at all.
For example, can I take those 3 yoshinoya and make a complete one with them?

I also want to know why 3 entries like that can happen? They are all pushed by the same guy, are they coming from the app directly? I mean like someone mapping them in the app/website.

Thank you

**Comment by bhousel** [comment]
> I also want to know why 3 entries like that can happen? They are all pushed by the same guy, are they coming from the app directly? I mean like someone mapping them in the app/website.

Most of the entries in this index were harvested from OpenStreetMap.  I looked up those entries for Yoshinoya and here's what I found:
* `yoshinoya-ffe475` is for Yoshinoya in these places: `["id", "my", "ph", "sg", "us"]` where the name uses latin characters
* `yoshinoya-20e6f1` is for Yoshinoya in these places: `["cn", "hk", "jp", "tw"]` where the name is in kanji characters
* `527fda-658eea` was probably a duplicate that got added when we processed the OpenStreetMap planet file.  
It was added because some one is adding a lot of `brand=吉野家 YOSHINOYA` in Japan only.  (I figured this out by using [nsi.guide](https://nsi.guide/index.html?t=brands&k=amenity&v=fast_food&tt=yoshi) and [Overpass Turbo](https://overpass-turbo.eu/s/139x))

What we can do to merge the duplicate is add a `matchName` property on `yoshinoya-20e6f1` to also recognize the name "吉野家 YOSHINOYA".  Then the scripts that build the index from collected strings will recognize that string as an alternate spelling, and not a new entry in the index.

So sometimes a duplicate is ok and expected because it is just indicating a regional tagging preference - other times a duplicate is not wanted because it is an inconsistency in how people are tagging features in OpenStreetMap.

**Reply by bhousel** [reply]
I made this change to remove the duplicate "吉野家 YOSHINOYA" in 42e2bd30

**Reply by SteaceP** [reply]
Thanks for that, this completely answer my questions.

Sorry for the delay, for some reason, I didn't receive a notification about your answer.
